Std/Data/Bool.lean

```diff
@@ -29,17 +29,21 @@ instance (p : Bool → Prop) [inst : DecidablePred p] : Decidable (∃ x, p x) :
   | _, isTrue hf => isTrue ⟨_, hf⟩
   | isFalse ht, isFalse hf => isFalse fun | ⟨true, h⟩ => absurd h ht | ⟨false, h⟩ => absurd h hf
 
-instance : LE Bool := leOfOrd
-instance : LT Bool := ltOfOrd
-instance : Max Bool := maxOfLe
-instance : Min Bool := minOfLe
+instance : LE Bool := ⟨(!. || .)⟩
```

Comment:
Do you think this is better or worse as follows?

```suggestion
instance : LE Bool := ⟨fun x y => x = true → y = true⟩
```
This feels like it might be nicer for proving than the current `(!x || y) = true`, and has the nice bonus that it's defeq to the LE instance on propositions.

Reply:
Personally, I think this is worse because the "bonus" relies on a non-trivial defeq. I don't think anyone should rely on `≤` being an implication under the hood, that's just weird!

Ultimately, I don't think there is much difference for code except for a tiny ms to elaborate the decision procedure for `->` when a decidable instance is required. But that's in elab and not in running code so it doesn't matter that much for performance.

I'm not against your version, I think it's clever! But I also think it is weird and confusing!

Reply:
PS: It's trivial to write a lemma that transforms `x ≤ y` into `x -> y`, so I question the necessity of having these defeq.

Reply:
In the end this probably doesn't matter that muxh; the important thing is getting the min and max fields right, which you already have. Unless @digama0 feels strongly, I'm ok with leaving this as is.